packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -86,10 +86,105 @@ const usePublicTaxonomies = () => {
 	}, [ taxonomies ] );
 };
 
+function usePostTypeNeedsUniqueIdentifier( publicPostTypes ) {
+	const postTypeLabels = useMemo( () =>
+		publicPostTypes?.reduce( ( accumulator, { labels } ) => {
+			const singularName = labels.singular_name.toLowerCase();
+			accumulator[ singularName ] =
+				( accumulator[ singularName ] || 0 ) + 1;
+			return accumulator;
+		}, {} )
+	);
+	return useCallback(
+		( { labels, slug } ) => {
+			const singularName = labels.singular_name.toLowerCase();
+			return postTypeLabels[ singularName ] > 1 && singularName !== slug;
+		},
+		[ postTypeLabels ]
+	);
+}
+
+export function usePostTypeArchiveMenuItems() {
+	const publicPostTypes = usePublicPostTypes();
```

Comment:
We have an issue here because we also need to take into account the [has_archive](https://developer.wordpress.org/reference/functions/register_post_type/)  and [REST API doesn't provide ](https://developer.wordpress.org/rest-api/reference/post-types/#schema)that info.. We need to update the schema and add it.